src/DDTrace/Util/HexConversion.php

```diff
@@ -6,6 +6,22 @@ final class HexConversion
 {
     public static function idToHex($id)
     {
-        return str_pad(dechex($id), 16, "0", STR_PAD_LEFT);
+        $hex = dd_trace_dec_hex((string) $id);
+        return str_pad($hex, 16, "0", STR_PAD_LEFT);
+    }
+
+    public static function hexToInt($hex)
+    {
+        if (dechex(hexdec($hex)) != dd_trace_dec_hex(dd_trace_hex_dec($hex))) {
```

Comment:
this seems expensive. Is it possible a cheaper check? at least it is possible to preserve `dd_trace_hex_dec($hex)` in 50% of the calls (assuming good id distribution).

Reply:
Thanks, made this a much cheaper and direct check.